### ISSUE TITLE
2.0 Has Broken Backgrounds

### ISSUE THREAD

**Issue by Sklore** [Sklore/HL_DD_5e_Colab#555]
The change below has impacted the Community Pack. We will need to align with the new standard. 

NOTE: Prevously, proficiencies with tools granted by races, backgrounds, etc. were handled by assigning ProfTool tags to the pick, which were then forwarded to the hero. This method is no longer supported. The new way of handling tool proficiencies is to bootstrap the tool from the pick with the Proficienc.Tool tag assigned on the bootstrap. Existing proficiencies will need to be converted over to the new method. Otherwise they won't show up on the printed character sheet or the specials tab in Hero Lab.

**Comment by Sklore** [comment]
From @shadowchemosh 

Not sure you saw my post about maybe waiting on changing everything. I wanted to try a couple Proof of Concept design ideas that would allow Tool Proficiency to be changed.

In other words you could add a Background and a end-user would be able to remove the Tools and replace with different ones.

But any solutions would require "how" the bootstrap happened to be changed. Meaning if you change everything over and then we have to change again its allot of double work. Maybe better to wait and and do the change "once".

**Comment by shadowchemosh** [comment]
Hopefully I can do a couple of quick tests tonight to see what will work best.  I really wish LW would stop flip flopping back and forth between bootstrapping and not bootstrapping.  Sigh...

**Comment by shadowchemosh** [comment]
If anyone has time to try out a new idea and let me know there thoughts.  I changed the "Charlatan" background to use a different method than bootstraps.  This method allows the tools "Forgery Kits" and "Disguise Kit" to be removed and changed.

The "bad" side is that changing the background will NOT remove the tools that get added.  They have to be removed manually.

The good side is that a gamer can "easily" change the tools.  The bad is that changing backgrounds is a little more work.

Let me know your thoughts....

**Comment by LordMorin** [comment]
I hope to look at the changes tomorrow for my own backgrounds that grant
tools, and my Adjustment that allows Tools to show the character's bonus
with them. Just finished my Tuesday night DM's session . . . :)

**Comment by TheEnthusiast** [comment]
I'm not sure if what you want is possible. Afaik, you cannot bootstrap through scripting, and once you bootstrap something, it cannot be removed, only hidden. I had this problem when I implemented the new ranger and had to remove the multiattack thing from the pet. 

Following the logic above, you'd have to bootstrap all the options and hide/unhide them according to what the user selected... Not a nice option.

**Comment by shadowchemosh** [comment]
@TheEnthusiast based on your comment I have to take that you did NOT try out what I did.  Your information is only about 50% correct.

Please actually try out what I did above and see how it works and if that is the approach you guys would like to go.

**Comment by TheEnthusiast** [comment]
@shadowchemosh I did try the new Charlatan. But I did not understand what you intend. There's no tool proficiency being added, nor tools on the gear tab on my hero lab.

For the matter, I created the artificer class, from the Jan/09/2017 Unearthed Arcana issue, implementing the tool proficiencies by using the bootstrap+Proficienc.Tool tag. There's no tool showing up in the user's gear tabs. But the tool proficiency shows on the skills tab.

However, my hero lab installation is crashing a LOT since the 8.0a update, so I'm not sure if I did somethin g wrong, or is my installation needing a complete reinstall. 

Anyone else want to try?

**Comment by LordMorin** [comment]
Tool proficiencies shouldn't show up on the gear tab unless you've added
the tool manually (aka bought or started with it.

What I've been finding is if a tool has been added correctly, and you are
proficient with it, then it shows up on the skills tab where tools have
always been. They now show you Proficiency bonus. I'd created an Adjustment
that did that originally; now I can get rid of it. It's no longer needed,
its build it.

My Expertise Adjustment is now longer working however. When I am over this
flu bug I'll have to look at how they are appyign Double Proficiency and
fix my Adjustments and other stuff.

A question though (I may be missing something due to medicine haze) but why
are you bootstrapping tools instead of just adding them as usual via tool
selection???

**Comment by shadowchemosh** [comment]
Let start with what I understand of Backgrounds.  They provide different proficiency to characters in the form of tools and skills.

The complaint from gamers I saw was that the Skills/Tools proficiencies was hard-coded and could not be changed.  I added adjustments to add, remove, change the skills because those are just tags.  Easy to do with a script.

But LW changed the Tools proficiencies to now be "bootstrapped" meaning they where hard-coded.  My change is to "auto-add" them not bootstrap them. Meaning the Tool proficiencies that are added/display on the "Skills" tab can be changed.  They can be changed by clicking the "X" and removing them and adding different ones instead.  You will notice they are not green colored which is how a bootstrapped Pick is displayed in the UI.

So maybe I am confused but I thought I was fixing the in-ability to change the Tool proficiencies that get added.  

Please note if we need the same "feature" for equipment I can add the equipment but was not sure if the equipment mentioned in the background is something that is suppose to be given for free.  This means the equipment gets auto-added to the character but it can still be removed/sold if the gamer wants.

Does that put us all on the same page?

**Comment by LordMorin** [comment]
I must be fuzzier than I thought . . .

You don't just use the tools and skills selections on each background?

I am lazy and if a background automatically assigns say 2 skills, I just
set the number of skills to 2 and limit them to the 2 they have access to.

I apologize if I am being an idiot . . .

**Comment by LordMorin** [comment]
First time I've looked at my background I've created. Has the Background
Tool Proficiencies button in the Upper Right under the Description window
always been there or is it new??

**Comment by shadowchemosh** [comment]
@LordMorin 

> A question though (I may be missing something due to medicine haze) but why are you bootstrapping tools instead of just adding them as usual via tool selection???

I am not bootstrapping "Tools" (ie Gear) LW requires that Tool Proficiency are bootstrapped now instead of being a tag.  What I did was "auto-add" the Tool Proficiency allowing them to be changed on the "Skills" tab if a person wants.

My comment about "equipment" was that I could do the same "auto-add" for Equipment.  In example a background gives you a free Belt Pouch.  I can setup the Belt Pouch to auto-add.  This means the Belt Pouch is added when you select the background and will display in the Gear tab.  BUT THE IMPORTANT PART IS IT IS NOT BOOTSTRAPPED.  This means the Belt Pouch can be sold/removed like any normal gear item you purchased/added on the Gear Tab.

My writing skills so suck. =(  Hope I made stuff more clear and not more confusing. =)

**Comment by LordMorin** [comment]
Once I am feeling better, and not thinking in a fog, I have to fix my own
custom backgrounds and I'll see more clearly what you are doing.

**Comment by Sklore** [comment]
Finally got around to testing. Sorry... RW keeps me bloody busy these days. 

Adding them appears to work fine. They print well. My concerns is this means we will do things differently to the SRD pack which means there will be two processes that exist for backgrounds. 

Removing them will drive confusion and complaints I suspect. It could get messy if you need to add an adjustment to remove the tools should you go through and add backgrounds incorrectly. 

I wonder what the reasoning for the change was.

**Comment by Sklore** [comment]
I honestly think this should go back to LW as this is in the SRD. 

**Proficiencies**
Each background gives a character proficiency in two skills (described in “Using Ability Scores”). In addition, most backgrounds give a character proficiency with one or more tools (detailed in “Equipment”). If a character would gain the same proficiency from two different sources, he or she can choose a different proficiency of the same kind (skill or tool) instead.

**Comment by Sklore** [comment]
Have emailed LW for their input on this one.

**Comment by Sklore** [comment]
Fyi, LW will add way to handle the tool proficiency duplication next release so we just need to worry about getting these fixed per standard process. 

That's something we need to automatically handle in the data files - the good news is I think it'll be relatively easy to do (i.e. prompt the user to add an additional tools proficiency if they have a duplicate). We'll schedule this for inclusion in the next release.

Thanks,

Colen

**Comment by TheEnthusiast** [comment]
Hi,

I'd like one of you to make a test. Its a small issue. Create an Artificer, Rogue, or anything that gives you thieves tools under the new method. You'll see the proficiency in skills, as usual.
Now mouse over your current load (Basics, under your speed). Does ti shows you carrying a thieves tools? If you check on Gear, there will be nothing.

**Comment by LordMorin** [comment]
Yep, it does. And it shouldn't until you add Thieves' Tools to your gear.

Did my test with a Rogue.

**Comment by Sklore** [comment]
Ok i think I caught them all.
